src/reports/components/report-sections/full-rule-header.tsx

```diff
@@ -19,15 +19,15 @@ export type FullRuleHeaderDeps = {
 
 export type FullRuleHeaderProps = {
     deps: FullRuleHeaderDeps;
-    rule: RuleResult;
+    cardResult: CardRuleResult;
```

Comment:
Question: Why link this name with a particular UI implementation (i.e. Card in this case). Like why not pass a more generic name like `results` and then pass appropriate data when we initialize these variables.

Reply:
The main reason is `CardRuleResults` is part of the `CardsViewModel`. Being a view model, it's couple to the view components that use it. i.e. this is not a domain model, hence I think it's fine the name of the prop reflect this. It also reflects the type name. I did change the name of the property to `cardRuleResult` though (here and on other files.